src/objective/regression_obj.cu

```diff
@@ -187,6 +185,31 @@ class RegLossObj : public FitInterceptGlmLike {
         .Eval(io_preds);
   }
 
+  void InitEstimation(MetaInfo const& info, linalg::Vector<float>* base_score) const override {
+    linalg::Vector<float> out;
+    FitInterceptGlmLike::InitEstimation(info, &out);
+    if (std::abs(this->param_.scale_pos_weight - 1.0f) <= kRtEps) {
+      common::Mean(this->ctx_, out, base_score);
+      return;
+    }
+
+    // Special handling for the scale_pos_weight parameter
+    auto w = this->param_.scale_pos_weight;
+    auto m = info.labels.Shape(0);
+    auto h_s = out.HostView();
+
+    for (std::size_t i = 0, n = h_s.Size(); i < n; ++i) {
+      // revert the mean back to sum, which is the number of positive samples
+      auto n_pos = h_s(i) * m;
```

Comment:
If it has sample weights, here it would need to use the sum of weights instead of the number of rows.